docs/architecture.md

```diff
@@ -1,188 +1,260 @@
+or a `ValidatingWebhookConfiguration` object with Kubernetes.
+
+<!--TODO:@vicuad points out we need a paragraph about the Audit Scanner.-->
+
+The diagram shows the architecture of a cluster running the Kubewarden stack:
+
+<!--TODO:To be clear. The stack refers to the entire diagram rather that the
+stacked components for e.g., CAP.-->
 
 ![Full architecture](/img/architecture.png)
 
-## Journey of a Kubewarden policy
+## The journey of a Kubewarden policy
+
+The architecture diagram appears complex at first. This section covers it step by step.
+
+### Default policy server
 
-The architecture diagram from above can be intimidating at first, this
-section explains it step by step.
+On a new cluster, the Kubewarden components defined are:
 
-### Default Policy Server
+<!--TODO:How are CRDs shown in the diagram-->
 
-On a fresh new cluster, the Kubewarden components defined are its Custom
-Resource Definitions, the `kubewarden-controller` Deployment and a `PolicyServer`
-Custom Resource named `default`.
+- the Custom Resource Definitions (CRD)
+- the `kubewarden-controller` Deployment
+- a `PolicyServer` Custom Resource named `default`.
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_01.png)
 
-`kubewarden-controller` notices the default `PolicyServer` resource and, as a result of that,
+<!--TODO:For the next paragraph. What type of architecture diagram is this? In the
+diagram, which is `policy-server`. The aquamarine box with 'Kubewarden Policy
+Server'? Is 'Deployment' capitalization correct? What does 'notices' mean
+exactly? For the diagram, what's the difference between solid box, dashed box,
+white box, aquamarine box? What are the arrows showing, should they be
+labelled?-->
+
+When the `kubewarden-controller` notices the default `PolicyServer` resource,
 it creates a Deployment of the `policy-server` component.
 
-As stated above, Kubewarden works as a Kubernetes Admission Webhook. Kubernetes
-dictates that all the Webhook endpoints must be secured with TLS.
-`kubewarden-controller` takes care of setting up this secure communication
-by doing these steps:
+Kubewarden works as a Kubernetes Admission Webhook.
+Kubernetes specifies using TLS to secure all Webhook endpoints.
+The `kubewarden-controller` sets up this secure communication
+by:
+
+1. Generating a self-signed Certificate Authority
+1. Use this CA to generate a TLS certificate key for the `policy-server` Service.
+
+<!--The capitalization on Secret is ok? Maybe in ``?-->
 
-1. Generate a self-signed Certificate Authority
-1. Use this CA to generate a TLS certificate and a TLS key for the
-   `policy-server` Service.
+These objects are all stored as Secret resources in Kubernetes.
 
-All these objects are stored into Kubernetes as Secret resources.
+<!--TODO:Next paragraph. `policy-server` is in the aquamarine box, right? I
+don't think the K8s ClusterIP Service is shown in the diagram anywhere right?
+Should it be?-->
```

Comment:
Yes, the service is not shown in the diagram. Maybe we can add it when we improve the diagram as a whole

Reply:
I think it's a good idea to add the k8s apiserver. Right now it is included in the "Kubernetes control plane" box.
 In the future, when showing the `audit-scanner` component, it will be obvious that normal requests go from the k8s apiserver to the webhooks, to the policy-server, and back. Meanwhile the audit-scanner doesn't receive requests.